src/mono/mono/utils/mono-threads.c

```diff
@@ -648,10 +644,6 @@ unregister_thread (void *arg)
 
 	mono_thread_info_suspend_unlock ();
 
-#ifdef HOST_BROWSER
-	mono_threads_wasm_on_thread_detached ();
-#endif
-
```

Comment:
without this, if some third party C code attaches to the runtime and does some managed stuff, and then the pthread dies without calling some Mono API to detach from the runtime, we won't get a notification about it.  Not sure if that's what you want.

Reply:
thanks, I will  look at it again